thirdeye-ui/src/app/pages/welcome-onboard-datasource-select-datasets/welcome-onboard-datasource-select-datasets.component.tsx

```diff
@@ -0,0 +1,235 @@
+        <FormControl component="fieldset" margin="dense">
+            <FormControlLabel
+                control={
+                    <Checkbox
+                        checked={checked}
+                        {...(indeterminate && { indeterminate })}
+                        color="primary"
+                        inputProps={{ "aria-label": name }}
+                        name={name}
+                        onChange={onChange}
+                    />
+                }
+                label={
+                    <>
+                        {labelPrimaryText}
+                        {labelSecondaryText ? (
+                            <FormHelperText>
+                                {labelSecondaryText}
+                            </FormHelperText>
+                        ) : null}
+                    </>
+                }
+            />
+        </FormControl>
+    );
+};
+
+export const WelcomeSelectDatasets: FunctionComponent = () => {
+    const { selectedDatasourceName, selectedDatasets, setSelectedDatasets } =
+        useOutletContext<WelcomeSelectDatasetOutletContext>();
+
+    const [datasource, setDatasource] = useState<Datasource | null>(null);
+    const [datasourceStatus, setDatasourceStatus] = useState<ActionStatus>(
+        ActionStatus.Initial
+    );
+
+    const { notify } = useNotificationProviderV1();
+    const { t } = useTranslation();
+
+    const handleToggleCheckbox = useCallback(
+        (event: React.ChangeEvent<HTMLInputElement>): void => {
+            const { name, checked } = event.target;
+            if (!name) {
+                return;
+            }
+            if (name === SELECT_ALL) {
+                setSelectedDatasets((selectedStateProp) => {
+                    const areAllCheckedState = !Object.values(
+                        selectedStateProp
+                    ).some((v) => !v);
+                    const newState = Object.assign(
+                        {},
+                        ...Object.keys(selectedStateProp).map((k) => ({
+                            [k]: !areAllCheckedState,
+                        }))
+                    );
+
+                    return newState;
+                });
+
+                return;
+            }
+
+            setSelectedDatasets((selectedStateProp) => ({
+                ...selectedStateProp,
+                [name]: checked,
+            }));
+        },
+        []
+    );
+
+    useEffect(() => {
+        if (selectedDatasourceName) {
+            setDatasourceStatus(ActionStatus.Working);
+            getDatasourceByName(selectedDatasourceName)
+                .then((datasourceProp: Datasource) => {
+                    setDatasource(datasourceProp);
+                    setDatasourceStatus(ActionStatus.Done);
+                })
+                .catch(() => {
+                    setDatasourceStatus(ActionStatus.Error);
+                });
+        } else {
+            setDatasourceStatus(ActionStatus.Error);
+        }
```

Comment:
We should create an action for this in `src/app/rest/datasources/datasources.actions.ts` if it does not exist already

Reply:
Created a hook for this. Supports both `name` and `id`

Reply:
Split into two hooks, one for each param type